src/IssueViz.UnitTests/Models/AnalysisIssueVisualizationTests.cs

```diff
@@ -0,0 +1,137 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.VisualStudio.Text;
+using Moq;
+using SonarLint.VisualStudio.Core.Analysis;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+
+namespace SonarLint.VisualStudio.IssueVisualization.UnitTests.Models
+{
+    [TestClass]
+    public class AnalysisIssueVisualizationTests
```

Comment:
This is a blunt copy-paste of AnalysisIssueLocationVisualizationTests, so I think your original suggestion of creating a PrimaryLocation property on AnalysisIssueVisualization was better than implementing the interface.
That, or maybe we should have a separate interface IAnalysisIssueSecondaryLocationVisualization

Reply:
Not sure; this is test code, though, so it probably doesn't check.

Reply:
Ah, again with this limitation :/